src/log_surgeon/finite_automata/PrefixTree.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef LOG_SURGEON_FINITE_AUTOMATA_PREFIX_TREE_HPP
+#define LOG_SURGEON_FINITE_AUTOMATA_PREFIX_TREE_HPP
+
+#include <cstdint>
+#include <optional>
+#include <stdexcept>
+#include <vector>
+
+namespace log_surgeon::finite_automata {
+/**
+ * Represents a prefix tree that stores all data needed by the TDFA registers.
+ *
+ * Each path from the root to an index represents a sequence of matched tag positions.
+ */
+class PrefixTree {
+public:
+    using id_t = uint32_t;
+    using position_t = int32_t;
+
+private:
+    /**
+     * Represents a prefix tree node. A node stores a potential value for a TDFA register.
+     *
+     * A node stores the current position at which a tag was matched, as well as the index of the
+     * prefix tree node corresponding to the previous match of the same tag.
+     *
+     * Note: A value of m_position < 0 indicates that the tag is currently unmatched in the lexed
```

Comment:
```suggestion
    * NOTE: A value of `m_position` < 0 indicates that the tag is currently unmatched in the lexed
```
1. We should use `` to surround member variables
2. We should use capital `NOTE`

Reply:
Removing this docstring as it is redundant with the `PrefixTree` docstring now.

Reply:
Changed.